src/main/java/org/tron/core/services/NodeInfoService.java

```diff
@@ -0,0 +1,177 @@
+  private Manager dbManager;
+
+  public NodeInfo getNodeInfo() {
+    NodeInfo nodeInfo = new NodeInfo();
+    setConnectInfo(nodeInfo);
+    setMachineInfo(nodeInfo);
+    setNodeInfo(nodeInfo);
+    setBlockInfo(nodeInfo);
+    return nodeInfo;
+  }
+
+  private void setMachineInfo(NodeInfo nodeInfo) {
+    MachineInfo machineInfo = new MachineInfo();
+    machineInfo.setThreadCount(threadMXBean.getThreadCount());
+    machineInfo.setCpuCount(Runtime.getRuntime().availableProcessors());
+    machineInfo.setTotalMemory(operatingSystemMXBean.getTotalPhysicalMemorySize());
+    machineInfo.setFreeMemory(operatingSystemMXBean.getFreePhysicalMemorySize());
+    machineInfo.setCpuRate(operatingSystemMXBean.getSystemCpuLoad());
+    machineInfo.setJavaVersion(runtimeMXBean.getSystemProperties().get("java.version"));
+    machineInfo
+        .setOsName(operatingSystemMXBean.getName() + " " + operatingSystemMXBean.getVersion());
+    machineInfo.setJvmTotalMemoery(memoryMXBean.getHeapMemoryUsage().getMax());
+    machineInfo.setJvmFreeMemory(
+        memoryMXBean.getHeapMemoryUsage().getMax() - memoryMXBean.getHeapMemoryUsage().getUsed());
+    machineInfo.setProcessCpuRate(operatingSystemMXBean.getProcessCpuLoad());
+    List<MemoryDescInfo> memoryDescInfoList = new ArrayList<>();
+    List<MemoryPoolMXBean> pools = ManagementFactory.getMemoryPoolMXBeans();
+    if (CollectionUtils.isNotEmpty(pools)) {
+      for (MemoryPoolMXBean pool : pools) {
+        MemoryDescInfo memoryDescInfo = new MemoryDescInfo();
+        memoryDescInfo.setName(pool.getName());
+        memoryDescInfo.setInitSize(pool.getUsage().getInit());
+        memoryDescInfo.setUseSize(pool.getUsage().getUsed());
+        memoryDescInfo.setMaxSize(pool.getUsage().getMax());
+        memoryDescInfo.setUseRate(pool.getUsage().getUsed() / pool.getUsage().getCommitted());
+        memoryDescInfoList.add(memoryDescInfo);
+      }
+    }
+    machineInfo.setMemoryDescInfoList(memoryDescInfoList);
+    //dead lock thread
+    long[] deadlockedIds = threadMXBean.findDeadlockedThreads();
+    if (ArrayUtils.isNotEmpty(deadlockedIds)) {
+      machineInfo.setDeadLockThreadCount(deadlockedIds.length);
+      ThreadInfo[] deadlockInfos = threadMXBean.getThreadInfo(deadlockedIds);
+      List<DeadLockThreadInfo> deadLockThreadInfoList = new ArrayList<>();
+      for (ThreadInfo deadlockInfo : deadlockInfos) {
+        DeadLockThreadInfo deadLockThreadInfo = new DeadLockThreadInfo();
+        deadLockThreadInfo.setName(deadlockInfo.getThreadName());
+        deadLockThreadInfo.setLockName(deadlockInfo.getLockName());
+        deadLockThreadInfo.setLockOwner(deadlockInfo.getLockOwnerName());
+        deadLockThreadInfo.setBlockTime(deadlockInfo.getBlockedTime());
+        deadLockThreadInfo.setWaitTime(deadlockInfo.getWaitedTime());
+        deadLockThreadInfo.setState(deadlockInfo.getThreadState().name());
+        deadLockThreadInfo.setStackTrace(deadlockInfo.getStackTrace().toString());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use "Arrays.toString(array)" instead. ([squid:S2116](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2116))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1615&issues=AWaF_zqnP8lmccBDp_1i&open=AWaF_zqnP8lmccBDp_1i)

[ISSUE_KEY]: AWaF_zqnP8lmccBDp_1i